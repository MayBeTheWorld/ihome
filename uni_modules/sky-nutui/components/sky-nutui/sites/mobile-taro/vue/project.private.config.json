{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}