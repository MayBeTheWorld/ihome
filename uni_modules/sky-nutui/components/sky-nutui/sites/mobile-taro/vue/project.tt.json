{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
