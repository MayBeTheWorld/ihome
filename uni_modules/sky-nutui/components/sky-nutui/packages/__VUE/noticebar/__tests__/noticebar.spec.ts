import { config, mount } from '@vue/test-utils';
import { nextTick, ref, reactive } from 'vue';
import NoticeBar from '../index.vue';
import NutIcon from '../../icon/index.vue';

function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

beforeAll(() => {
  config.global.components = {
    NutIcon
  };
});

afterAll(() => {
  config.global.components = {};
});

Object.defineProperty(window.HTMLElement.prototype, 'clientWidth', {
  value: 375
});

test('close event', async () => {
  const wrapper = mount(NoticeBar, {
    props: {
      text: '华为畅享9新品即将上市，活动期间0元预约可参与抽奖，赢HUAWEI WATCH等好礼，更多产品信息请持续关注！',
      direction: 'across',
      closeMode: true,
      color: 'red',
      background: 'green',
      delay: 1
    }
  });
  const closeDom = wrapper.find('.right-icon');
  closeDom.trigger('click');
  expect(wrapper.emitted('close')).toBeTruthy();
  wrapper.setProps({
    text: '123'
  });
  await nextTick();
  const content = wrapper.find('.content');
  expect(content.html()).toContain('123');
});

test('slot event', async () => {
  const wrapper = mount(NoticeBar, {
    slots: {
      default: () => 'Custom Content'
    }
  });
  await nextTick();
  const content = wrapper.find('.content');
  expect(content.html()).toContain('Custom Content');
});

test('icon custom', async () => {
  const wrapper = mount({
    components: {
      'nut-noticebar': NoticeBar
    },
    template: `
        <nut-noticebar
        left-icon="https://img13.360buyimg.com/imagetools/jfs/t1/72082/2/3006/1197/5d130c8dE1c71bcd6/e48a3b60804c9775.png"
        :scrollable="false"
      >
        <a href="https://www.jd.com">京东商城</a>
      </nut-noticebar>
    `
  });
  await nextTick();
  expect(wrapper.html()).toMatchSnapshot();
});

test('vertical scroll', async () => {
  const wrapper = mount({
    components: {
      'nut-noticebar': NoticeBar
    },
    template: `
      <nut-noticebar direction='vertical'  :list="horseLamp" ></nut-noticebar>
    `,
    setup() {
      const horseLamp = ref(['惊喜红包免费领', '爆款准点秒', '买超值优惠', '赢百万京豆']);
      return { horseLamp };
    }
  });
  await nextTick();
  expect(wrapper.html()).toMatchSnapshot();
});

test('custon scroll list', async () => {
  const wrapper = mount({
    components: {
      'nut-noticebar': NoticeBar
    },
    template: `
      <nut-noticebar direction='vertical' :height='50' :speed='10' :standTime='1000' :list="[]" >
        <div class="custom-item" v-for="(item,index) in horseLamp" :key="item">{{item}}</div>
      </nut-noticebar>
    `,
    setup() {
      const horseLamp = ref(['惊喜红包免费领', '爆款准点秒', '买超值优惠', '赢百万京豆']);
      return { horseLamp };
    }
  });
  await nextTick();
  expect(wrapper.html()).toMatchSnapshot();
});
