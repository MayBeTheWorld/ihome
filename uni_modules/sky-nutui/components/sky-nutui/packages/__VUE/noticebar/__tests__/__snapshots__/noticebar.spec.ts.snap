// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custon scroll list 1`] = `
"<view class=\\"nut-noticebar\\">
  <!--v-if-->
  <view class=\\"nut-noticebar-vertical\\" style=\\"height: 50px;\\">
    <view class=\\"horseLamp_list\\">
      <div class=\\"custom-item\\" style=\\"height: 50px; line-height: 50px;\\">惊喜红包免费领</div>
      <div class=\\"custom-item\\" style=\\"height: 50px; line-height: 50px;\\">爆款准点秒</div>
      <div class=\\"custom-item\\" style=\\"height: 50px; line-height: 50px;\\">买超值优惠</div>
      <div class=\\"custom-item\\" style=\\"height: 50px; line-height: 50px;\\">赢百万京豆</div>
    </view>
    <view class=\\"go\\">
      <!--v-if-->
    </view>
  </view>
</view>"
`;

exports[`icon custom 1`] = `
"<view class=\\"nut-noticebar\\">
  <view class=\\"nut-noticebar-page\\">
    <view class=\\"left-icon\\" style=\\"background-image: url(https://img13.360buyimg.com/imagetools/jfs/t1/72082/2/3006/1197/5d130c8dE1c71bcd6/e48a3b60804c9775.png);\\">
      <!--v-if-->
    </view>
    <view class=\\"wrap\\">
      <view class=\\"content\\" style=\\"animation-delay: 1s; animation-duration: 0s; transform: translateX(0);\\"><a href=\\"https://www.jd.com\\">京东商城</a></view>
    </view>
    <!--v-if-->
  </view>
  <!--v-if-->
</view>"
`;

exports[`vertical scroll 1`] = `
"<view class=\\"nut-noticebar\\">
  <!--v-if-->
  <view class=\\"nut-noticebar-vertical\\" style=\\"height: 40px;\\">
    <ul class=\\"horseLamp_list\\">
      <li class=\\"horseLamp_list_item\\">惊喜红包免费领</li>
      <li class=\\"horseLamp_list_item\\">爆款准点秒</li>
      <li class=\\"horseLamp_list_item\\">买超值优惠</li>
      <li class=\\"horseLamp_list_item\\">赢百万京豆</li>
    </ul>
    <view class=\\"go\\">
      <!--v-if-->
    </view>
  </view>
</view>"
`;
